src/build_tools/vs_util.py

```diff
@@ -33,8 +33,23 @@
 import pathlib
 import subprocess
 import sys
-from typing import Union
+from typing import Union, Optional
 
+def find_vswhere() -> Optional[pathlib.Path]:
+  path = pathlib.Path(r'C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe')
+  return path if path.exists() else None
+
+def get_vcvarsall_from_vswhere(vswhere_path: pathlib.Path) -> Optional[pathlib.Path]:
+  result = subprocess.run([str(vswhere_path), '-find', 'VC/Auxiliary/Build/vcvarsall.bat', '-utf8'], stdout=subprocess.PIPE)
+  first = result.stdout.splitlines()[0].decode()
+  if first == '':
+    return None
+  return pathlib.Path(first)
+
+def get_display_name_from_vswhere(vswhere_path: pathlib.Path) -> Optional[str]:
+  result = subprocess.run([str(vswhere_path), '-property', 'displayName', '-utf8'], stdout=subprocess.PIPE)
```

Comment:
Unless it’s absolutely necessary, I prefer to set `shell=False` to avoid any unnecessary complexities coming from `cmd.exe`.